pkg/operator/operator_test.go

```diff
@@ -206,59 +219,24 @@ func TestOperatorSyncClusterAPIControllerHealthCheckControllerNotDeployed(t *tes
 		},
 	}
 
-	if err := op.syncClusterAPIController(oc); err != nil {
-		t.Errorf("Failed to sync machine API controller: %v", err)
-	}
-
-	d, err := kubeclientSet.AppsV1().Deployments(targetNamespace).Get(deploymentName, metav1.GetOptions{})
-	if err != nil {
-		t.Errorf("Failed to get %q deployment: %v", deploymentName, err)
-	}
-
-	if deploymentHasContainer(d, hcControllerName) {
-		t.Errorf("Did not expect to find %q container", hcControllerName)
-	} else {
-		t.Logf("%q container not found as expected", hcControllerName)
-	}
-}
-
-func TestOperatorSyncClusterAPIControllerHealthCheckControllerDeployed(t *testing.T) {
-	fg := &v1.FeatureGate{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: MachineAPIFeatureGateName,
-		},
-		Spec: v1.FeatureGateSpec{
-			FeatureSet: configv1.TechPreviewNoUpgrade,
-		},
-	}
-
-	kubeclientSet := fakekube.NewSimpleClientset()
-	op, err := initOperator(fg, kubeclientSet)
-	if err != nil {
-		t.Errorf("Unable to init operator: %v", err)
-	}
-
-	oc := OperatorConfig{
-		TargetNamespace: targetNamespace,
-		Controllers: Controllers{
-			Provider:           "controllers-provider",
-			NodeLink:           "controllers-nodelink",
-			MachineHealthCheck: "controllers-machinehealthcheck",
-		},
-	}
+	for _, tc := range tests {
+		kubeclientSet := fakekube.NewSimpleClientset()
+		op, err := initOperator(tc.featureGate, kubeclientSet)
+		if err != nil {
+			t.Errorf("Unable to init operator: %v", err)
```

Comment:
If this errors, what is the value of `op` a few lines below? Presumably it can be `nil` - so should this be `Fatalf`?